exercises/practice/strain/.meta/Example.vb

```diff
@@ -0,0 +1,24 @@
+Imports System
+Imports System.Collections.Generic
+Imports System.Runtime.CompilerServices
+
+Public Module Strain
+    <Extension()>
+    Public Function KeepMethod(Of T)(ByVal collection As IEnumerable(Of T), ByVal predicate As Func(Of T, Boolean)) As IEnumerable(Of T)
+        Return ApplyPredicate(collection, predicate)
+    End Function
+
+    <Extension()>
+    Public Function DiscardMethod(Of T)(ByVal collection As IEnumerable(Of T), ByVal predicate As Func(Of T, Boolean)) As IEnumerable(Of T)
+        Return ApplyPredicate(collection, Function(x) Not predicate(x))
+    End Function
+
+    Private Function ApplyPredicate(Of T)(ByVal collection As IEnumerable(Of T), ByVal predicate As Func(Of T, Boolean)) As IEnumerable(Of T)
+        ' could knock this down to a simple LINQ expression but restriction prevents that
+        Dim filtered = New List(Of T)()
+        For Each item In collection
+            If predicate(item) Then filtered.Add(item)
```

Comment:
```suggestion
    Private Function ApplyPredicate(Of T)(ByVal coll As IEnumerable(Of T), ByVal pred As Func(Of T, Boolean)) As IEnumerable(Of T)
        ' could knock this down to a simple LINQ expression but restriction prevents that
        Dim filtered = New List(Of T)()
        For Each item In coll
            If pred(item) Then filtered.Add(item)
```

Collection and Predicate are case-insensitive keywords, so best not to use them for variable names.